docs/reference/api/connectivity/networksocket/DNS.md

```diff
@@ -0,0 +1,89 @@
+## DNS
+
+The DNS provides an interface to do DNS host name resolutions.
```

Comment:
We probably need to invent some name for this API.
It sound a bit weird to tell "DNS provides interface for DNS"

Reply:
Name is now: DNS resolver